### PR TITLE
Support Binary/BeGuid type properties

### PR DESCRIPTION
**Pull request by medikornov**
Closes https://github.com/iTwin/imodel-native/issues/201
itwinjs-core PR: https://github.com/iTwin/itwinjs-core/pull/5452
itwinjs-core: https://github.com/iTwin/itwinjs-core/tree/presentation/support-guid-type-properties

### CODE REVIEW

File: iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp [hunk 1]
```diff
@@ -49,11 +49,35 @@ struct CommonRulesEngineSymbolsProvider : IECSymbolProvider
         evalResult.SetValueList(*IValueListResult::Create(values));
         return ExpressionStatus::Success;
         }
+
+    /*---------------------------------------------------------------------------------**//**
+    * @bsimethod
+    +---------------+---------------+---------------+---------------+---------------+------*/
+    static ExpressionStatus GuidToStr(EvaluationResult& evalResult, void*, EvaluationResultVector& args)
```
Comment (grigasp) [inline iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp:56]: Please add unit tests for this

File: iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp [hunk 2]
```diff
@@ -49,11 +49,35 @@ struct CommonRulesEngineSymbolsProvider : IECSymbolProvider
...
+
+        EvaluationResultCR arg = args[0];
+        if (!arg.IsECValue() && !arg.GetECValue()->IsBinary())
+            ExpressionStatus::IncompatibleTypes;
+
+        size_t guidSize = sizeof(BeGuid);
+        BeGuid* guid = (BeGuid*)arg.GetECValue()->GetBinary(guidSize);
+
+        if (!guid->IsValid())
+            return ExpressionStatus::IncompatibleTypes;
```
Comment (grigasp) [inline iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp:70]: No sure this is right. Need a test that tries to convert an empty guid to string to confirm.

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp [hunk 3]
```diff
@@ -953,6 +953,194 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, DescriptorOverride_FiltersB
...
 
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+DEFINE_SCHEMA(Guid_AppliesExtendedDataRule, R"*(
+    <ECEntityClass typeName="A">
+        <ECProperty propertyName="GuidProp" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, Guid_AppliesExtendedDataRule)
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp:964]: I think I'd just add to `AppliesContentItemExtendedDataFromPresentationRules` test for testing extended data.

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp [hunk 4]
```diff
@@ -953,6 +953,194 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, DescriptorOverride_FiltersB
...
+    instanceGuid2.FromString("814f3e14-63f2-4511-89a8-43ff3b527492");
+    IECInstancePtr instance2 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [instanceGuid2](IECInstanceR instance) {
+        instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid2, sizeof(BeGuid)));
+        });
+    KeySetPtr input = KeySet::Create(bvector<IECInstancePtr>{instance1, instance2});
+
+    // create the rule set
+    PresentationRuleSetPtr rules = PresentationRuleSet::CreateInstance(BeTest::GetNameOfCurrentTest());
+    m_locater->AddRuleSet(*rules);
+    Utf8StringCR instanceFilter = Utf8PrintfString("this.GuidProp = StrToGuid(\"%s\")", instanceGuid2.ToString().c_str());
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp:1035]: Can't take reference of an rvalue. Instead do this:
```cpp
Utf8PrintfString instanceFilter("this.GuidProp = StrToGuid(\"%s\")", instanceGuid2.ToString().c_str());
```

The tests below also have this issue.

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp [hunk 5]
```diff
@@ -953,6 +953,194 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, DescriptorOverride_FiltersB
...
+
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+DEFINE_SCHEMA(Guid_FilteringUsingSpecificationsFilter, R"*(
+    <ECEntityClass typeName="A">
+        <ECProperty propertyName="GuidProp" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, Guid_FilteringUsingSpecificationsFilter)
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp:1017]: I'd mode the test next to `ContentInstancesOfSpecificClasses_InstanceFilter` and rename to `ContentInstancesOfSpecificClasses_InstanceFilter_UsingGuidProperty`

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp [hunk 6]
```diff
@@ -953,6 +953,194 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, DescriptorOverride_FiltersB
...
+
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+DEFINE_SCHEMA(Guid_FilteringUsingDescriptorsFieldsFilter, R"*(
+    <ECEntityClass typeName="A">
+        <ECProperty propertyName="GuidProp" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, Guid_FilteringUsingDescriptorsFieldsFilter)
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp:1059]: I'd move the test next to `DescriptorOverride_FiltersByNavigationProperty` and rename to `DescriptorOverride_FiltersByGuidProperty`

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp [hunk 7]
```diff
@@ -953,6 +953,194 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, DescriptorOverride_FiltersB
...
+
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+DEFINE_SCHEMA(Guid_FilteringUsingDescriptorsInstanceFilter, R"*(
+    <ECEntityClass typeName="A">
+        <ECProperty propertyName="GuidProp" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, Guid_FilteringUsingDescriptorsInstanceFilter)
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp:1104]: I'd move the test to `Content_Filtering.cpp` and rename to `DescriptorInstanceFilter_FiltersByGuidProperty`.

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp [hunk 8]
```diff
@@ -9737,6 +9925,76 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, LoadsPrimitiveArrayProperty
...
+    ContentDescriptorCPtr descriptor = GetValidatedResponse(m_manager->GetContentDescriptor(AsyncContentDescriptorRequestParams::Create(s_project->GetECDb(), rules->GetRuleSetId(), RulesetVariables(), nullptr, 0, *KeySet::Create())));
+    ASSERT_TRUE(descriptor.IsValid());
+    ASSERT_EQ(1, descriptor->GetVisibleFields().size());
+    rapidjson::Document expectedFieldType;
+    expectedFieldType.Parse(Utf8PrintfString(R"({
+        "ValueFormat": "Array",
+        "TypeName": "binary[]",
+        "MemberType": {
+            "ValueFormat": "Primitive",
+            "TypeName": "binary"
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp:9971]: I wonder why extended type name wasn't carried over as `TypeName` like in `SetsTypeNameAsExtendedTypeNameIfProvided` test - can you check please?

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp [hunk 9]
```diff
@@ -9737,6 +9925,76 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, LoadsPrimitiveArrayProperty
...
 
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+DEFINE_SCHEMA(LoadsGuidArrayPropertyValue, R"*(
+    <ECEntityClass typeName="MyClass">
+        <ECArrayProperty propertyName="GuidsArrayProperty" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, LoadsGuidArrayPropertyValue)
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp:9936]: I think we're missing a `LoadsGuidPropertyValue` test

**Review by grigasp — CHANGES_REQUESTED** [review]

File: iModelCore/ECPresentation/Source/Content/ContentDescriptorBuilder.cpp [hunk 10]
```diff
@@ -393,12 +393,20 @@ struct ContentPropertiesAppender : ContentSpecificationsHandler::PropertyAppende
...
+        // don't support binary (not guid) and igeometry properties
         // note: we don't want to these fields even if the above says we should..
-        if (ecProperty.GetIsPrimitive() && (PRIMITIVETYPE_Binary == ecProperty.GetAsPrimitiveProperty()->GetType() || PRIMITIVETYPE_IGeometry == ecProperty.GetAsPrimitiveProperty()->GetType()))
+        if (ecProperty.GetIsPrimitive())
             {
-            DIAGNOSTICS_DEV_LOG(DiagnosticsCategory::Content, LOG_TRACE, "Binary and IGeometry fields are not supported - skip.");
-            return false;
+            if (PRIMITIVETYPE_Binary == ecProperty.GetAsPrimitiveProperty()->GetType() && ecProperty.GetAsPrimitiveProperty()->GetExtendedTypeName() != EXTENDED_TYPENAME_BeGuid)
+                {
+                DIAGNOSTICS_DEV_LOG(DiagnosticsCategory::Content, LOG_TRACE, "Binary (not Guid) field is not supported - skip.");
```
[the remark this answers is not in the document]
Comment (medikornov) [reply iModelCore/ECPresentation/Source/Content/ContentDescriptorBuilder.cpp]: fixed

File: iModelCore/ECPresentation/Source/Content/ContentDescriptorBuilder.cpp [hunk 11]
```diff
@@ -393,12 +393,20 @@ struct ContentPropertiesAppender : ContentSpecificationsHandler::PropertyAppende
...
+                return false;
+                }
+            if (PRIMITIVETYPE_IGeometry == ecProperty.GetAsPrimitiveProperty()->GetType())
+                {
+                DIAGNOSTICS_DEV_LOG(DiagnosticsCategory::Content, LOG_TRACE, "IGeometry field is not supported - skip.");
```
[the remark this answers is not in the document]
Comment (medikornov) [reply iModelCore/ECPresentation/Source/Content/ContentDescriptorBuilder.cpp]: fixed

File: iModelCore/ECPresentation/Source/Content/ContentItemBuilder.cpp [hunk 12]
```diff
@@ -97,7 +97,8 @@ rapidjson::Document ContentValuesFormatter::GetFallbackArrayValue(ArrayECPropert
         else if (prop.GetIsPrimitiveArray())
             {
             PrimitiveType primitiveType = prop.GetAsPrimitiveArrayProperty()->GetPrimitiveElementType();
-            json.PushBack(GetFallbackPrimitiveValue(prop, primitiveType, value, &json.GetAllocator()), json.GetAllocator());
+            Utf8String extendedType = prop.GetAsPrimitiveArrayProperty()->GetExtendedTypeName();
```
[the remark this answers is not in the document]
Comment (medikornov) [reply iModelCore/ECPresentation/Source/Content/ContentItemBuilder.cpp]: fixed

File: iModelCore/ECPresentation/Source/Hierarchies/NavigationQuery.cpp [hunk 13]
```diff
@@ -57,8 +57,9 @@ void NavigationQueryExtendedData::AddRangesData(ECPropertyCR prop, PropertyGroup
         {
         rapidjson::Value range(rapidjson::kObjectType);
         PrimitiveType type = prop.GetAsPrimitiveProperty()->GetType();
-        range.AddMember("from", ValueHelpers::GetJsonFromString(type, rangeSpec->GetFromValue(), &GetAllocator()), GetAllocator());
-        range.AddMember("to", ValueHelpers::GetJsonFromString(type, rangeSpec->GetToValue(), &GetAllocator()), GetAllocator());
+        Utf8String extendedType = prop.GetAsPrimitiveProperty()->GetExtendedTypeName();
```
[the remark this answers is not in the document]
Comment (medikornov) [reply iModelCore/ECPresentation/Source/Hierarchies/NavigationQuery.cpp]: fixed

File: iModelCore/ECPresentation/Source/Shared/ValueHelpers.cpp [hunk 14]
```diff
@@ -327,11 +330,17 @@ rapidjson::Document ValueHelpers::GetJsonFromStructValue(ECStructClassCR structC
                 doc.AddMember(propertyNameJson, GetJsonFromStructValue(*v.GetColumnInfo().GetStructType(), v, &doc.GetAllocator()), doc.GetAllocator());
                 break;
             case ValueKind::VALUEKIND_Array:
-                doc.AddMember(propertyNameJson, GetJsonFromArrayValue(v, &doc.GetAllocator()), doc.GetAllocator());
+                {
+                Utf8StringCR extendedType = v.GetColumnInfo().GetProperty()->GetIsPrimitiveArray() ? v.GetColumnInfo().GetProperty()->GetAsPrimitiveArrayProperty()->GetExtendedTypeName() : "";
```
[the remark this answers is not in the document]
Comment (medikornov) [reply iModelCore/ECPresentation/Source/Shared/ValueHelpers.cpp]: Probably understood. Fixed

File: iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.cpp [hunk 15]
```diff
@@ -464,7 +464,15 @@ static void AssertInstanceValueValid(IECInstanceCR instance, RapidJsonValueCR va
     ECValue instanceValue;
     ECObjectsStatus status = instance.GetValue(instanceValue, propertyName);
     EXPECT_EQ(ECObjectsStatus::Success, status);
-    EXPECT_EQ(instanceValue, GetECValueFromJson(value, *property));
+
+    if (instanceValue.IsBinary() && property->GetIsPrimitive() && property->GetAsPrimitiveProperty()->GetExtendedTypeName() == EXTENDED_TYPENAME_BeGuid)
+        {
+        size_t sizeOfGuid = sizeof(BeGuid);
+        BeGuid* guid = (BeGuid*)instanceValue.GetBinary(sizeOfGuid);
+        EXPECT_EQ(ECValue(guid->ToString().c_str()), GetECValueFromJson(value, *property));
```
[the remark this answers is not in the document]
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.cpp]: fiixed

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/Content_DistinctValuesTests.cpp [hunk 16]
```diff
@@ -501,6 +501,55 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, GetDistinctPrimitiveValues)
...
+        rules->GetRuleSetId(), RulesetVariables(), "", 0, *KeySet::Create())));
+    PagedDataContainer<DisplayValueGroupCPtr> values = GetValidatedResponse(m_manager->GetDistinctValues(AsyncDistinctValuesRequestParams::Create(s_project->GetECDb(),
+        *descriptor, std::make_unique<PropertiesContentFieldMatcher>(*classA->GetPropertyP("Prop"), RelatedClassPath()))));
+
+    // validate
+    ASSERT_EQ(2, values.GetSize());
+
+    EXPECT_STREQ(instanceGuid1.ToString().c_str(), values[0]->GetDisplayValue().c_str());
+    EXPECT_EQ(1, values[0]->GetRawValues().size());
+
```
[the remark this answers is not in the document]
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/Content_DistinctValuesTests.cpp]: removed

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Hierarchies/Hierarchies_InstanceFiltering.cpp [hunk 17]
```diff
@@ -1649,6 +1649,91 @@ TEST_F(RulesDrivenECPresentationManagerNavigationTests, InstanceFiltering_Filter
...
+    IECInstancePtr a13 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [&](IECInstanceR instance){instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid1, sizeof(BeGuid))); instance.SetValue("FilterProp", ECValue((Byte*)&instanceGuid2, sizeof(BeGuid))); });
+    IECInstancePtr a21 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [&](IECInstanceR instance){instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid2, sizeof(BeGuid))); instance.SetValue("FilterProp", ECValue((Byte*)&instanceGuid1, sizeof(BeGuid))); });
+    IECInstancePtr a22 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [&](IECInstanceR instance){instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid2, sizeof(BeGuid))); instance.SetValue("FilterProp", ECValue((Byte*)&instanceGuid2, sizeof(BeGuid))); });
+    IECInstancePtr a3 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [&](IECInstanceR instance){instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid3, sizeof(BeGuid))); instance.SetValue("FilterProp", ECValue((Byte*)&instanceGuid1, sizeof(BeGuid))); });
```
[the remark this answers is not in the document]
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Integration/Hierarchies/Hierarchies_InstanceFiltering.cpp]: fixed

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Hierarchies/Hierarchies_InstanceFiltering.cpp [hunk 18]
```diff
@@ -1649,6 +1649,91 @@ TEST_F(RulesDrivenECPresentationManagerNavigationTests, InstanceFiltering_Filter
...
+TEST_F(RulesDrivenECPresentationManagerNavigationTests, InstanceFiltering_FiltersLabelGroupingNodesByGuid)
+    {
+    // dataset
+    ECClassCP classA = GetClass("A");
+    BeGuid instanceGuid1;
+    instanceGuid1.FromString("2d75965d-46ef-480a-adb8-42276dbb66a9");
+    BeGuid instanceGuid2;
+    instanceGuid2.FromString("814f3e14-63f2-4511-89a8-43ff3b527492");
+    BeGuid instanceGuid3;
+    instanceGuid3.FromString("182238d2-e836-4640-9b40-38be6ca49623");
```
[the remark this answers is not in the document]
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Integration/Hierarchies/Hierarchies_InstanceFiltering.cpp]: fixed

File: iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp (hunk 1 above)
Comment (medikornov) [reply iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp:56]: Added unit tests when binary property is a `valid guid`, `invalid guid` and `null`

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp (hunk 3 above)
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp]: fixed

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp (hunk 4 above)
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp]: fixed

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp (hunk 8 above)
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp]: Fixed. Check `ECSchemaHelper::GetTypeName`

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp (hunk 9 above)
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp:9899]: Added

File: iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.cpp [hunk 19]
```diff
@@ -868,6 +868,13 @@ void RulesEngineTestHelpers::CacheNode(IHierarchyCacheR cache, Utf8StringCR conn
     cache.Cache(node, dsInfo.GetIdentifier(), 0, NodeVisibility::Visible);
     }
 
+BeGuid RulesEngineTestHelpers::CreateGuidFromString(Utf8CP str)
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.cpp:871]: missing function header

File: iModelCore/ECPresentation/Source/Shared/ValueHelpers.cpp [hunk 20]
```diff
@@ -363,13 +360,17 @@ rapidjson::Document ValueHelpers::GetJsonFromArrayValue(IECSqlValue const& sqlVa
...
             case ValueKind::VALUEKIND_Array:
-                {
-                Utf8String extendedArrayType = v.GetColumnInfo().GetProperty()->GetIsPrimitiveArray() ? v.GetColumnInfo().GetProperty()->GetAsPrimitiveArrayProperty()->GetExtendedTypeName() : "";
-                doc.PushBack(GetJsonFromArrayValue(v, &doc.GetAllocator(), extendedArrayType), doc.GetAllocator());
+                doc.PushBack(GetJsonFromArrayValue(v, &doc.GetAllocator()), doc.GetAllocator());
                 break;
-                }
             case ValueKind::VALUEKIND_Primitive:
-                doc.PushBack(GetJsonFromPrimitiveValue(v.GetColumnInfo().GetDataType().GetPrimitiveType(), v, &doc.GetAllocator(), extendedType), doc.GetAllocator());
+                ECPropertyCP property = v.GetColumnInfo().GetOriginProperty();
```
Comment (grigasp) [inline iModelCore/ECPresentation/Source/Shared/ValueHelpers.cpp:366]: Generally, if a function returns a pointer, it means it may return a `nullptr` (otherwise it would return a reference). Unless you're 100% sure that it's not going to do that, it's better to just add a null check:
```cpp
if (property == nullptr)
    DIAGNOSTICS_HANDLE_FAILURE(DiagnosticsCategory::Default, "<error message here, ideally with some information about the column>");
```

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/Content_Filtering.cpp [hunk 21]
```diff
@@ -395,6 +395,56 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, DescriptorInstanceFilter_Us
...
+    {
+    ECClassCP classA = GetClass("A");
+    BeGuid instanceGuid1 = BeGuid(true);
+    auto guid1 = instanceGuid1.ToString();
+    IECInstancePtr instance1 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [instanceGuid1](IECInstanceR instance)
+        {
+        instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid1, sizeof(BeGuid)));
+        });
+    BeGuid instanceGuid2 = BeGuid(true);
+    auto guid2 = instanceGuid2.ToString();
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/Content_Filtering.cpp:416]: looks like `guid1` and `guid2` are unused

**Review by grigasp — CHANGES_REQUESTED** [review]
almost there

File: iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.cpp (hunk 19 above)
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.cpp:874]: fixed

File: iModelCore/ECPresentation/Source/Shared/ValueHelpers.cpp (hunk 20 above)
Comment (medikornov) [reply iModelCore/ECPresentation/Source/Shared/ValueHelpers.cpp:366]: fixed?

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/Content_Filtering.cpp (hunk 21 above)
Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/Content_Filtering.cpp]: removed

**Review by grigasp — APPROVED** [review]

File: iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp [hunk 22]
```diff
@@ -195,19 +196,22 @@ struct NodeECInstanceContextEvaluator : PropertySymbol::ContextEvaluator
         }
     ExpressionContextPtr _GetContext() override
         {
+        if (nullptr != m_instanceContext.get())
```
Comment (saskliutas) [inline iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp:199]:
```suggestion
        if (m_instanceContext.IsValid())
```

Comment (medikornov) [reply iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp]: fixed

File: iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp [hunk 23]
```diff
@@ -967,12 +967,12 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, DescriptorOverride_FiltersB
     BeGuid instanceGuid1 = BeGuid(true);
     IECInstancePtr instance1 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [instanceGuid1](IECInstanceR instance) 
         {
-        instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid1, sizeof(BeGuid)));
+        instance.SetValue("GuidProp", ECValue((const Byte*)&instanceGuid1, sizeof(BeGuid)));
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp:970]: Our convention is to use `Byte const*` rather than `const Byte*`.

Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp]: fixed
